src/Shared/ExceptionHandling.cs

```diff
@@ -41,7 +45,15 @@ static ExceptionHandling()
         /// <returns></returns>
         private static string GetDebugDumpPath()
         {
-            string debugPath = Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH");
+            string debugPath =
+#if CLR2COMPATIBILITY || MICROSOFT_BUILD_ENGINE_OM_UNITTESTS
```

Comment:
@Forgind
Made this change after you signed off. ChangeWaves are not visible to the MICROSOFT_BUILD_ENGINE_OM_UNITTESTS assembly so I had to ifdef it out. The alternative is to move ChangeWaves from Microsoft.Build.Framework into shared sources, but that seems like a bigger change than this one.

Reply:
Maybe move the test to Engine.UnitTests? It kinda makes sense that Engine.OM.UnitTests wouldn't be able to see Framework, but I'm not a huge fan of ifdef'ing based on being in a test—to me, it feels like making tests not actually test code, just test similar code. I'll let @BenVillalobos opine on the right way to handle ChangeWaves visibility, though.

Reply:
The entire test assembly ends up depending on ChangeWaves so, not just a few tests, so it's not feasible to move a few tests. Only other solution that I see is to move ChangeWaves into the shared sources directory.

Reply:
I remember having issues in trying to make changewaves shared. Lots of "This type exists multiple times" issues. If you can move it with no issues then by all means. It's unrelated to what this PR is doing so I'd say make an issue for it but don't let it block.

Reply:
Add a comment and this is fine by me.